src/helper/isImplementedCache.ts

```diff
@@ -10,10 +10,17 @@ export class IsImplementedCache {
 
   constructor(private cacheDuration: number) {}
 
-  get(eventName: string): ImplementedEvent | undefined {
-    const event: ImplementedEvent | undefined = this.cache.find(e => {
-      return e.name === eventName;
-    });
+  get(eventName: string, skillId?: string): ImplementedEvent | undefined {
+    let event: ImplementedEvent | undefined;
+    if (skillId) {
+      event = this.cache.find(e => {
+        return e.name === eventName && e.skillId === skillId;
+      });
+    } else {
+      event = this.cache.find(e => {
+        return e.name === eventName;
+      });
+    }
```

Comment:
This part can be handled in one line of code, avoiding the if-else statement.

```javascript
event = this.cache.find(e => { return skillId ? e.name === eventName && e.skillId === skillId: e.name === eventName });
```

Reply:
resolved